runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -6684,9 +6684,26 @@ done:;
 				if (interfaceClass == iTable->interfaceClass) {
 					receiverClass->lastITable = iTable;
 foundITableCache:
-					_sendMethod = *(J9Method**)((UDATA)receiverClass + ((UDATA*)(iTable + 1))[methodIndex]);
+					if (J9_UNEXPECTED(J9_ARE_ANY_BITS_SET(methodIndexAndArgCount, J9_ITABLE_INDEX_TAG_BITS))) {
+						/* Object or private interface method invoke */
+						if (J9_ARE_ANY_BITS_SET(methodIndexAndArgCount, J9_ITABLE_INDEX_METHOD_INDEX)) {
+							if (J9_ARE_ANY_BITS_SET(methodIndexAndArgCount, J9_ITABLE_INDEX_OBJECT)) {
+								/* Object method not in the vTable */
+								_sendMethod = J9VMJAVALANGOBJECT_OR_NULL(_vm)->ramMethods + methodIndex;
+							} else {
+								/* Private interface method */
+								_sendMethod = interfaceClass->ramMethods + methodIndex;
+							}
+						} else {
+							/* Object method in the vTable */
+							_sendMethod = *(J9Method**)((UDATA)receiverClass + methodIndex);
+						}
+					} else {
+						/* Standard interface method */
+						_sendMethod = *(J9Method**)((UDATA)receiverClass + ((UDATA*)(iTable + 1))[methodIndex]);
+					}
 					romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(_sendMethod);
-					if (!J9_ARE_ANY_BITS_SET(romMethod->modifiers, J9AccPublic)) {
+					if (J9_ARE_NO_BITS_SET(romMethod->modifiers, J9AccPublic | J9AccPrivate)) {
```

Comment:
This should only be applicable to JDK 11 +.  Discussed maybe not resolving the private case for older releases.

Reply:
Will be done in the resolve.